magenta-lib/src/main/scala/magenta/deployment_type/Param.scala

```diff
@@ -38,21 +40,24 @@ case class Param[T](
       (DeploymentPackage, DeployTarget) => Either[String, T]
     ] = None,
     deprecatedDefault: Boolean = false
-)(implicit register: ParamRegister) {
+)(implicit register: ParamRegister, reads: Reads[T], manifest: Manifest[T]) {
```

Comment:
As explained below, it's better to add the requirement for the `Reads[T]` to the _constructor_ of the `Param[T]` class, so that once the `Param[T]` is constructed, it always parses things in the same way, rather than depending on how it's called.